src/LanguageServer.php

```diff
@@ -206,6 +206,13 @@ public function initialize(ClientCapabilities $capabilities, string $rootPath =
                 // Find composer.json
                 if ($this->composerJson === null) {
                     $composerJsonFiles = yield $this->filesFinder->find(Path::makeAbsolute('**/composer.json', $rootPath));
+
+                    // If we sort our findings by string length (shortest to longest),
+                    // the first entry will be the project's root composer.json.
```

Comment:
Good call, should have thought about it for more than two seconds.  I think `substr_count($a, '/')` will do it correctly?